mongo/client_side_encryption_examples_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func Example_clientSideEncryption() {
+	// This would have to be the same master key that was used to create the encryption key
+	localKey := make([]byte, 96)
+	if _, err := rand.Read(localKey); err != nil {
+		log.Fatal(err)
+	}
+	kmsProviders := map[string]map[string]interface{}{
+		"local": {
+			"key": localKey,
+		},
+	}
+	keyVaultNamespace := "admin.datakeys"
+
+	uri := "mongodb://localhost:27017"
+	autoEncryptionOpts := options.AutoEncryption().
+		SetKeyVaultNamespace(keyVaultNamespace).
+		SetKmsProviders(kmsProviders)
+	clientOpts := options.Client().ApplyURI(uri).SetAutoEncryptionOptions(autoEncryptionOpts)
```

Comment:
Small difference between this example and the corresponding Java one: I added a `ApplyURI` call because these languages are supposed to be copy/pasteable and I assumed that meant they should be runnable as well. @jyemin I'm not sure if the URI was left out of the Java ones on purpose. If so, I can remove it here if needed.